ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -395,29 +404,46 @@ public void addDoubleStats(ColumnStatisticsData cd, List<String> ls) {
     ls.add(lowValue + dc.getLowValue() + "'");
   }
 
+  public String checkHistogram(ColumnStatisticsData cd) {
+    byte[] buffer = null;
+
+    if (cd.isSetDoubleStats() && cd.getDoubleStats().isSetHistogram()) {
```

Comment:
should we check other types as well?

Reply:
Good catch, we need to handle all the other supported data types here, I have added that